src/BuildScriptGeneratorCli/Commands/DockerfileCommand.cs

```diff
@@ -95,5 +93,66 @@ internal override bool IsValidInput(IServiceProvider serviceProvider, IConsole c
 
             return true;
         }
+
+        internal override IServiceProvider GetServiceProvider(IConsole console)
+        {
+            // Gather all the values supplied by the user in command line
+            SourceDir = string.IsNullOrEmpty(SourceDir) ?
+                Directory.GetCurrentDirectory() : Path.GetFullPath(SourceDir);
+
+            // NOTE: Order of the following is important. So a command line provided value has higher precedence
+            // than the value provided in a configuration file of the repo.
+            var config = new ConfigurationBuilder()
+                .AddIniFile(Path.Combine(SourceDir, Constants.BuildEnvironmentFileName), optional: true)
+                .AddEnvironmentVariables()
+                .Add(GetCommandLineConfigSource())
+                .Build();
+
+            // Override the GetServiceProvider() call in CommandBase to pass the IConsole instance to
+            // ServiceProviderBuilder and allow for writing to the console if needed during this command.
+            var serviceProviderBuilder = new ServiceProviderBuilder(LogFilePath, console)
+                .ConfigureServices(services =>
+                {
+                    // Configure Options related services
+                    // We first add IConfiguration to DI so that option services like
+                    // `DotNetCoreScriptGeneratorOptionsSetup` services can get it through DI and read from the config
+                    // and set the options.
+                    services
+                        .AddSingleton<IConfiguration>(config)
+                        .AddOptionsServices()
+                        .Configure<BuildScriptGeneratorOptions>(options =>
+                        {
+                            // These values are not retrieve through the 'config' api since we do not expect
+                            // them to be provided by an end user.
+                            options.SourceDir = SourceDir;
+                            options.ScriptOnly = false;
```

Comment:
similar question: what is the difference between this sourcedir and the one above that we receive from user?